### PR TITLE
[IOTDB-3513] Avoid double-writing of the write ahead log for data under RatisConsensus

### PR DESCRIPTION
**Pull request by HeimingZ**
Disable wal when using ratis.

### CODE REVIEW

**Review by OneSizeFitsQuorum — COMMENTED** [review]
PTAL

File: server/src/main/java/org/apache/iotdb/db/service/DataNode.java [hunk 1]
```diff
@@ -281,6 +281,11 @@ private void setUp() throws StartupException, QueryProcessException {
     registerManager.register(CacheHitRatioMonitor.getInstance());
     registerManager.register(CompactionTaskManager.getInstance());
     JMXService.registerMBean(getInstance(), mbeanName);
+
+    // close wal when using ratis consensus
+    if (!config.getDataRegionConsensusProtocolClass().equals(ConsensusFactory.RatisConsensus)) {
```
[the remark this answers is not in the document]
Comment (HeimingZ) [reply server/src/main/java/org/apache/iotdb/db/service/DataNode.java]: fixed

File: server/src/main/java/org/apache/iotdb/db/engine/StorageEngineV2.java [hunk 2]
```diff
@@ -223,10 +224,12 @@ public void recover() {
     asyncRecover(recoveryThreadPool, futures);
 
     // wait until wal is recovered
-    try {
-      WALRecoverManager.getInstance().recover();
-    } catch (WALException e) {
-      logger.error("Fail to recover wal.", e);
+    if (!config.getDataRegionConsensusProtocolClass().equals(ConsensusFactory.RatisConsensus)) {
```
[the remark this answers is not in the document]
Comment (HeimingZ) [reply server/src/main/java/org/apache/iotdb/db/engine/StorageEngineV2.java]: fixed

File: server/src/main/java/org/apache/iotdb/db/engine/storagegroup/DataRegion.java [hunk 3]
```diff
@@ -304,8 +304,17 @@ public DataRegion(
     } else {
       lastFlushTimeManager = new LastFlushTimeManager();
     }
-    // recover tsfiles
-    recover();
+
+    // recover tsfiles unless consensus protocol is ratis and storage engine is not ready
+    if (config.getDataRegionConsensusProtocolClass().equals(ConsensusFactory.RatisConsensus)
```
[the remark this answers is not in the document]
Comment (HeimingZ) [reply server/src/main/java/org/apache/iotdb/db/engine/storagegroup/DataRegion.java]: fixed

**Review by HTHou — APPROVED** [review]

**Pull request merged**